src/providers/mod.rs

```diff
@@ -111,6 +115,13 @@ pub trait Provide {
         Err(ResponseStatus::PsaErrorNotSupported)
     }
 
+    /// Get the Provider ID the provider.
+    fn provider_id(&self) -> ProviderID;
+
+    /// Get an optional reference to the KeyInfoStore for this provider. Providers that do not
+    /// implement a key store should return None.
+    fn get_key_info_store(&self) -> Option<&KeyInfoStore>;
+
```

Comment:
The `Provide` trait should ideally only contain the actual operations that the providers can perform with the exception of when it would be really hard/unpractical to do otherwise like with the `describe` function above. But that one should maybe be in another trait!

My point is that, from the `CoreProvider`, you can directly get this information by adding a field containing a vector of all `ManageKeyInfo` implementation and use that trait to get the key information. You can add all of them to the `CoreProvider` during service building.

Reply:
Gotcha. Will change this too.

Reply:
This is where a `list_keys` implementation on every provider would've helped 🤡 `CoreProvider` would then just aggregate. 

Also, I'd rather go down the route of asking the provider to tell you which `ManageKeyInfo` it's using, because otherwise imagine having two different key info stores and switching between them at some point. The `ListKeys` would then report keys that you don't have access to - you see them because they're saved in a different key info manager, but the actual provider can't see them.